### PR TITLE
Contentful integration

### PR DESCRIPTION
**Pull request by nomad302**
Contentful API data fetched and displayed!!

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,11 +1,9 @@
...
+import Post from 'post/post';
 
-export default class App extends React.Component<IHelloProps>{
-    public render(): JSX.Element {
-        return (
-            <h1>{this.props.greet} To {this.props.company}</h1>
-        );
-    }
-}
+export const App: React.FunctionComponent<object> = (): JSX.Element => (
```
Comment (hrc7505) [inline src/App.tsx:5]: This should be export default because we are exporting only one module from this file.

Comment (nomad302) [reply src/App.tsx:5]: Okay, I Will do Thanks.

File: src/commonstyles.css [hunk 2]
```diff
@@ -0,0 +1,12 @@
+html,
+body {
+    width: 100%;
+    height: 100%;
+    margin: 0;
+    padding: 0;
+    font-family: Helvetica, sans-serif;
+}
+
+.app-container{
```
Comment (hrc7505) [inline src/commonstyles.css:10]: Create a separate file for component style. Every component should have it's own css file and should be imported in component's file. Only common style rules should be here.

File: src/post/Post.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        return (
+            <div className="post-container">
+                <h1>{postContents.title}</h1>
+                <h4>{postContents.description}</h4>
+                <img
+                    src={postContents.image.fields.file.url}
+                    alt={postContents.image.fields.description}
+                />
+                <div>
+                    <p>{postContents.paragraph}</p>
```
Comment (hrc7505) [inline src/post/Post.tsx:43]: It should be dynamic. You should fix this. The content type should have the field that shows text-editor where you can write lots of text and can format it too.  Here you can write only 4 paragraphs. What if we have more paragraphs?

**Review by hrc7505 — CHANGES_REQUESTED** [review]
I added few comments please fix those issues.

File: src/post/Post.tsx (hunk 3 above)
Comment (nomad302) [reply src/post/Post.tsx:43]: okay, will work on this, Thanks.

File: package.json [hunk 4]
```diff
@@ -26,9 +26,11 @@
     "webpack-dev-server": "^3.11.0"
   },
   "dependencies": {
+    "@contentful/rich-text-html-renderer": "^14.1.1",
```
Comment (hrc7505) [inline package.json:29]: Should be in devDependecies

File: package.json [hunk 5]
```diff
@@ -26,9 +26,11 @@
...
+    "@contentful/rich-text-types": "^14.1.1",
```
Comment (hrc7505) [inline package.json:30]: Should be in devDependecies

File: src/model/common/Iimage.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
+export interface IImageDimensions {
```
Comment (hrc7505) [inline src/model/common/Iimage.ts:1]: Filename is not proper.  Should be like `imageModels.ts`.

File: src/post/Post.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+                <h1>{postContents.title}</h1>
+                <h4>{postContents.description}</h4>
+                <img
+                    src={postContents.image.fields.file.url}
+                    alt={postContents.image.fields.description}
+                />
+                <section
+                        className="blogContent"
+                        dangerouslySetInnerHTML={{ __html: documentToHtmlString(postContents.body) }}
+                    />
```
Comment (hrc7505) [inline src/post/Post.tsx:44]: Formatting required

**Review by hrc7505 — CHANGES_REQUESTED** [review]
I added some comments please fix and resolve

**Review by hrc7505 — APPROVED** [review]
Now this PR looks better.

**Pull request merged**